src/test/java/core/basesyntax/DaoStorageTest.java

```diff
@@ -0,0 +1,103 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.DaoStorage;
+import core.basesyntax.dao.DaoStorageImpl;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.exception.DaoStorageException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class DaoStorageTest {
+    static final String FRUIT = "banana";
+    static final int QUANTITY = 100;
+    static final int NEGATIVE_QUANTITY = -10;
+    private static DaoStorage daoStorage;
+
+    @BeforeAll
+    public static void setUp() {
+        daoStorage = new DaoStorageImpl();
+    }
+
+    @AfterEach
+    public void clear() {
+        StorageImpl.getStorage().clear();
+    }
+
+    @Test
+    public void daoStorage_putArgumentsNullOrNegative_notOk() {
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.setNewValue(null, QUANTITY));
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.setNewValue(FRUIT, null));
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.setNewValue(FRUIT, NEGATIVE_QUANTITY));
+    }
+
+    @Test
+    public void daoStorage_mergeArgumentsNullOrNegative_notOk() {
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.concatenateValue(null, QUANTITY));
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.concatenateValue(FRUIT, null));
+        assertThrows(DaoStorageException.class,
+                () -> daoStorage.concatenateValue(FRUIT, NEGATIVE_QUANTITY));
+    }
+
+    @Test
+    public void daoStorage_getArgumentsNull_notOk() {
+        assertThrows(DaoStorageException.class, () -> daoStorage.getValue(null));
+    }
+
+    @Test
+    public void daoStorage_merge_Ok() {
+        assertFalse(StorageImpl.getStorage().containsKey(FRUIT));
+        daoStorage.concatenateValue(FRUIT, QUANTITY);
+        assertTrue(StorageImpl.getStorage().containsKey(FRUIT));
+        assertEquals(QUANTITY, StorageImpl.getStorage().get(FRUIT));
+
+        int expected = QUANTITY + QUANTITY;
+        daoStorage.concatenateValue(FRUIT, QUANTITY);
+        assertEquals(expected, StorageImpl.getStorage().get(FRUIT));
+    }
+
+    @Test
+    public void daoStorage_put_Ok() {
+        assertFalse(StorageImpl.getStorage().containsKey(FRUIT));
+        daoStorage.setNewValue(FRUIT, QUANTITY);
+        assertTrue(StorageImpl.getStorage().containsKey(FRUIT));
+        assertEquals(QUANTITY, StorageImpl.getStorage().get(FRUIT));
+
+        int expected = QUANTITY + QUANTITY;
+        daoStorage.setNewValue(FRUIT, QUANTITY + QUANTITY);
+        assertEquals(expected, StorageImpl.getStorage().get(FRUIT));
```

Comment:
please read about AAA unit test pattern and use it. It will improve readability of your tests